jaxrs-api/src/test/java/jakarta/ws/rs/core/CookieTest.java

```diff
@@ -57,4 +57,27 @@ public void testEquals() {
         assertTrue(newCookie.equals(newCookie1));
         assertFalse(newCookie.equals(newCookie2));
     }
+
+    /**
+     * Test of equals method, of class Cookie and NewCookie on instances built
+     * with builder.
+     */
+    @Test
+    public void testEqualsOnInstancesBuiltWithBuilder() {
```

Comment:
The claims this test verifies are actually not rather obvious to detect due to the style of the test code. I do not enforce BDD, but it would be good if we could specify what the test actually wants to achieve, so other programmers can later easily understand and extend them.

Reply:
Well again I agree with you but I did things the same way they are already inspired by existing test `CookieTest.testEquals()`.

Reply:
I know, and I am not proud of some things we did in the past, but let's try to make it even better together. :-)

Reply:
+1 I agree with you.

I personnaly think that a JAVA specification should encourage all best design and code practices that people can use as a model for their development and thats why I made this proposal. 
Because IMO these telescoping constructors are no longer a good design/coding practice to encourage at all. Let me know if you agree.

You said that you prefer to see better code quality for new APIs addition and again I agree with you. 
Actually it's the APIs addition done by @chkal if I'm right, about samesite concept that leads me to this proposal. This new addition was simple and works fine but it introduces a 7 parameters constructor. Do you guys want to encourage this ?

Reply:
Not unlimited. The sense of this specification is solely to define a common interface above already existing products, not to define a style of programming. We define just an API to use those products. JAX-RS is not a framework, not a product, not a language, and has _no particular_ style, neither a _preferred_ style. It must be long-living and cannot adjust to short-loving styles and fashions again and again. While we like builders today, we didn't in the past. Possibly we won't in the future. Certainly we try to _adjust_ builders where needed and useful. But it is **definitively not** our charter _to actively push_ programmers into _one particular_ programming style. That's completely up to the user of our API. Not vice versa.

Reply:
Well maybe "quality" was not the right world. IDK
But IMO whether or not an API is using telescoping contructors or builders is about code readability and maintainability.

By the way so that I know what is do you mean by "code quality" ?